public/css/timeline.less

```
@@ -65,9 +70,86 @@
       display: block;
       border-top: 1px solid black;
       position: absolute;
-      bottom: var(--stepRowHeight);
       right: 0;
       left: 0;
+      // -1 to exclude result row
+      top: ~"calc((var(--stepRowHeight) * calc(var(--primaryRows) - 1)) + var(--daysHeaderHeight))";
+    }
+
+    .timescale {
+      display: grid;
+      grid-template-columns: repeat(~"calc(var(--primaryColumns) * var(--timestampsPerDay))", minmax(0,1fr));
+      border-left: 1px solid @gray-lighter; // this is required to maintain the grid layout
+      grid-area: ~"4 / 2 / 4 / 3";
+
+      .ticks {
+        position: relative;
+        border-right: 1px solid @gray-lighter;
+        border-left: 1px solid @gray-lighter;
+
+        &:after { // overlaps the unnecessary part of border-left
+          content: '';
+          position: absolute;
+          top: 0.25em;
+          left: -1px; // overlap the border-left
+          right: 0;
+          bottom: 0;
+          background: @body-bg-color;
+        }
+      }
+
+      .timestamp {
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        margin-top: 0.5em;
+        padding-top: 0.5em;
+        font-size: .5em;
+        position: relative;
+        left: -50%;
+        line-height: 1;
+
+        &.midnight {
+          left: 50%;
+        }
+
+        > span:last-child {
+          opacity: 0.5;
+        }
+      }
+
+      span:nth-last-of-type(2), // last .ticks before .midnight
+      .midnight {
+        grid-area: ~"1 / -2 / 1 / -1";
+      }
+    }
+
+    .clock {
+      display: grid;
+      grid-template-columns: repeat(var(--primaryColumns), 1fr);
```

Same orientation, same rules. (As e.g. the header)

```suggestion
      grid-template-columns: repeat(var(--primaryColumns), minmax(var(--minimumPrimaryColumnWidth), 1fr));
```